From b530a2ba75caf0706d83dae50e86ed30dcdc83d6 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Sun, 5 Aug 2012 19:27:08 -0300
Subject: [PATCH 612/670] ENGR00223337-2 Don't mangle IRQ_VDDA_BRNOUT for MX23
 arch

This has been introduced on commit e93820b3 and makes the kernel to
fail to build for MX23 arch so we avoid the IRQ_VDDA_BRNOUT
manipulation for MX23 arch as this IRQ is not available on MX23.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@freescale.com>
---
 drivers/power/mxs/linux.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/power/mxs/linux.c b/drivers/power/mxs/linux.c
index d5185fa..0d6c729 100644
--- a/drivers/power/mxs/linux.c
+++ b/drivers/power/mxs/linux.c
@@ -1171,7 +1171,9 @@ static int __init mxs_bat_init(void)
 		disable_irq(IRQ_DCDC4P2_BRNOUT);
 		disable_irq(IRQ_BATT_BRNOUT);
 		disable_irq(IRQ_VDDD_BRNOUT);
+#ifndef CONFIG_ARCH_MX23
 		disable_irq(IRQ_VDDA_BRNOUT);
+#endif
 		if (no_battery)
 			disable_irq(IRQ_VDDIO_BRNOUT);
 #ifndef CONFIG_ARCH_MX28
@@ -1184,7 +1186,9 @@ static int __init mxs_bat_init(void)
 		mxs_set_irq_fiq(IRQ_DCDC4P2_BRNOUT, 1);
 		mxs_set_irq_fiq(IRQ_BATT_BRNOUT, 1);
 		mxs_set_irq_fiq(IRQ_VDDD_BRNOUT, 1);
+#ifndef CONFIG_ARCH_MX23
 		mxs_set_irq_fiq(IRQ_VDDA_BRNOUT, 1);
+#endif
 		if (no_battery)
 			mxs_set_irq_fiq(IRQ_VDDIO_BRNOUT, 1);
 #ifndef CONFIG_ARCH_MX28
@@ -1199,7 +1203,9 @@ static int __init mxs_bat_init(void)
 		enable_irq(IRQ_DCDC4P2_BRNOUT);
 		enable_irq(IRQ_BATT_BRNOUT);
 		enable_irq(IRQ_VDDD_BRNOUT);
+#ifndef CONFIG_ARCH_MX23
 		enable_irq(IRQ_VDDA_BRNOUT);
+#endif
 		if (no_battery)
 			enable_irq(IRQ_VDDIO_BRNOUT);
 #ifndef CONFIG_ARCH_MX28
-- 
1.8.0

